extensions/sql-migration/src/wizard/skuRecommendation/skuRecommendationPage.ts

```diff
@@ -0,0 +1,598 @@
+						const skuRecAutoRefreshTimeInMins = this.migrationStateModel.refreshGetSkuRecommendationFrequency / 60000;
+
+						this._skuDataCollectionTimerText.value = constants.AZURE_RECOMMENDATION_STATUS_AUTO_REFRESH_TIMER(Math.ceil(skuRecAutoRefreshTimeInMins - elapsedTimeInMins));
+					} else {
+						// Timer is only for first refresh.
+						this._skuDataCollectionTimerText.value = constants.AZURE_RECOMMENDATION_STATUS_MANUAL_REFRESH_TIMER;
+					}
+
+					await this._skuDataCollectionStatusContainer.updateCssStyles({ 'display': 'flex' });
+				}
+
+				else if (this.migrationStateModel.performanceCollectionStopped()) {
+					await this._skuDataCollectionStatusIcon.updateProperties({
+						iconPath: IconPathHelper.stop
+					});
+					this._skuDataCollectionStatusText.value = constants.AZURE_RECOMMENDATION_STATUS_STOPPED;
+					this._skuDataCollectionTimerText.value = '';
+				}
+				break;
+			}
+
+			case PerformanceDataSourceOptions.OpenExisting: {
+				if (utils.hasRecommendations(this.migrationStateModel)) {
+				}
+				break;
+			}
+
+			// initial state before "Get Azure recommendation" dialog
+			default: {
+				await this._skuDataCollectionStatusContainer.updateCssStyles({ 'display': 'none' });
+				break;
+			}
+		}
+
+		await this.refreshCardText(false);
+	}
+
+	public async refreshAssessment(): Promise<void> {
+		await this.startCardLoading();
+		this.migrationStateModel._runAssessments = true;
+		await this.constructDetails();
+	}
+
+	public async refreshAzureRecommendation(): Promise<void> {
+		await this.startCardLoading();
+		await this.migrationStateModel.getSkuRecommendations();
+
+		const skuRecommendationError = this.migrationStateModel._skuRecommendationResults?.recommendationError;
+		if (skuRecommendationError) {
+			this.wizard.message = {
+				text: constants.SKU_RECOMMENDATION_ERROR(this._serverName),
+				description: skuRecommendationError.message,
+				level: azdata.window.MessageLevel.Error
+			};
+		}
+
+		await this.refreshSkuRecommendationComponents();
+	}
+
+	public async onPageEnter(pageChangeInfo: azdata.window.WizardPageChangeInfo): Promise<void> {
+		this.wizard.registerNavigationValidator((pageChangeInfo) => {
+			this.wizard.message = { text: '' };
+			if (pageChangeInfo.newPage < pageChangeInfo.lastPage) {
+				return true;
+			}
+
+			const errors: string[] = [];
+			if (errors.length > 0) {
+				this.wizard.message = {
+					text: errors.join(EOL),
+					level: azdata.window.MessageLevel.Error
+				};
+				return false;
+			}
+			return true;
+		});
+		await this.constructDetails();
+		await this._skuDataCollectionToolbar._setToolbarState();
+		this.wizard.nextButton.enabled = this.migrationStateModel._assessmentResults !== undefined;
+	}
+
+	public async onPageLeave(pageChangeInfo: azdata.window.WizardPageChangeInfo): Promise<void> {
+		this.wizard.registerNavigationValidator(pageChangeInfo => true);
+		this.wizard.message = { text: '' };
+		// this.eventListener?.dispose();
```

Comment:
What about this?  Should this comment be removed or uncommented?

Reply:
removed